report/src/main/webapp/app/layout/main/url/reports/cookie.html

```diff
@@ -67,20 +67,18 @@
       </tr>
       </tbody>
     </table>
-
   </div>
 
-
-  <div
-      ng-if="(case.result.cookieName || case.result.name) && res.cookie.cookie[0].status == 'FAILED'">
-    Cookie with name: '{{case.result.cookieName}}' not found
-    <span
-        ng-if="case.result.cookieValue">' and value: '{{case.result.cookieValue}}' not found </span>
-  </div>
-  <div
-      ng-if="(case.result.cookieName || case.result.name) && res.cookie.cookie[0].status != 'FAILED'">
-    Cookie with name: '{{case.result.cookieName}}' found
-    <span ng-if="case.result.cookieValue">' and value: '{{case.result.cookieValue}}' found </span>
+  <div ng-if="case.comparator.parameters.action == 'test'">
+    <div ng-if="case.comparator.stepResult.status == 'FAILED'">
+      Cookie with name: <span class="status failed">{{case.result.cookieName}}</span>
+      <span ng-if="case.result.cookieValue"> and value:
+      <span class="status failed">{{case.result.cookieValue}}</span></span> not found
+    </div>
+    <div ng-if="case.comparator.stepResult.status != 'FAILED'">
+      Cookie with name: <span class="status passed">{{case.result.cookieName}}</span> <span ng-if="case.result.cookieValue">
+      and value: <span class="status passed">{{case.result.cookieValue}}</span></span> found
+    </div>
```

Comment:
Is it possible to avoid duplication in both cases: when it failed and when it succeeded? The only difference is `failed`/`passed` class and the word "not".

How about:

```
<div ng-if="case.comparator.parameters.action == 'test'" ng-class="case.comparator.stepResult.status == 'FAILED' ? 'failed' : 'passed'">
    Cookie with name: <span class="status">{{case.result.cookieName}}</span> 
    <span ng-if="case.result.cookieValue"> and value: <span class="status passed">{{case.result.cookieValue}}</span></span>
    <span ng-if="case.comparator.stepResult.status == 'FAILED'">not</span> found
</div>
```